examples/src/test/java/io/grpc/examples/routeguide/RouteGuideServerTest.java

```diff
@@ -52,26 +54,34 @@
  */
 @RunWith(JUnit4.class)
 public class RouteGuideServerTest {
+  /**
+   * This rule manages automatic graceful shutdown for the registered channel at the end of test.
+   */
+  @Rule
+  public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
   private RouteGuideServer server;
   private ManagedChannel inProcessChannel;
   private Collection<Feature> features;
 
   @Before
   public void setUp() throws Exception {
-    String uniqueServerName = "in-process server for " + getClass();
+    // Generate a unique in-process server name.
+    String serverName = InProcessServerBuilder.generateName();
     features = new ArrayList<Feature>();
-    // use directExecutor for both InProcessServerBuilder and InProcessChannelBuilder can reduce the
+    // Use directExecutor for both InProcessServerBuilder and InProcessChannelBuilder can reduce the
     // usage timeouts and latches in test. But we still add timeout and latches where they would be
     // needed if no directExecutor were used, just for demo purpose.
     server = new RouteGuideServer(
-        InProcessServerBuilder.forName(uniqueServerName).directExecutor(), 0, features);
+        InProcessServerBuilder.forName(serverName).directExecutor(), 0, features);
```

Comment:
I think we need to register the server for cleanups as well here.

Reply:
server is of type `RouteGuideServer`, not `io.grpc.Server`, and application has its own life-cycle management. So we don't need to register the server for cleanup. The same reason for the other comment about client.

Reply:
Got it. Then this is fine.